### PR TITLE
The Paco Fun Switch

### PR DESCRIPTION
**Pull request by Takykardi**
## About The Pull Request

If you absolutely, positively need to convert money into noise at eight hundred rounds per minute, you might need a fun switch. It's simpler to design self-loading firearms to be fully automatic to begin with, and then slow them down with a disconnector piece. An enterprising gunsmith realized that you could disable the Paco disconnector in the tunnels by just sticking a precisely-shaped piece of metal into the backplate, and started selling conversion kits. The Solarian department of Alcohol, Tobacco, and Firearms considered that "fun switch" to be the galaxy's smallest machinegun, so don't order one online... unless you're lonely and really want to hang out with an SATF squad.

Demonstrated: https://cdn.discordapp.com/attachments/848683007170969681/1021134110783189073/2022-09-18_14-00-59.mp4

## Why It's Good For The Game

Upon installation, the Paco receives the ability to fire fully automatic. As in, it's locked into firing fully automatic. This makes it hellish to control if the trigger is held for longer than a few seconds, but it's quite useful if you happen to be surrounded and need to put thirty-four holes in someone in under two seconds. Or at least in their general direction.

https://user-images.githubusercontent.com/60326959/190936900-520d8812-5902-4b9a-939c-84c06f977c23.mp4
Video is outdated, taken prior to the requested balance changes.
image
The single sprite addition.

## Changelog
```changelog
add: Added a gun "upgrade" for the FS .35 "Paco", the Fun Switch
```

### CODE REVIEW

File: strings/loot_data/all_spawn_data.txt [hunk 1]
```diff
@@ -2013,6 +2013,7 @@ paths    spawn_tags    blacklisted    spawn_value    price_tag    prob_all_accom
 /obj/item/weapon/gun_upgrade/mechanism/overshooter    obj,item,gun_upgrade    blacklisted=0    spawn_value=5    price_tag=100   100    
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow    obj,item,gun_upgrade    blacklisted=1    spawn_value=2    price_tag=100   100    
 /obj/item/weapon/gun_upgrade/mechanism/weintraub    obj,item,gun_upgrade    blacklisted=0    spawn_value=3.33333    price_tag=100   100    
+/obj/item/weapon/gun_upgrade/mechanism/funswitch    obj,item,gun_upgrade    blacklisted=0    spawn_value=3.33333    price_tag=100   100
```
Comment (Jamini) [inline strings/loot_data/all_spawn_data.txt:2016]: This is an automatically generated file and doesn't need to be changed for this PR.

File: strings/loot_data/all_spawn_data.txt [hunk 2]
```diff
@@ -6199,6 +6200,7 @@ paths    spawn_tags    blacklisted    spawn_value    spawn_price    prob_accompa
 /obj/item/weapon/gun_upgrade/mechanism/overshooter    obj;item;gun_upgrade;guns    blacklisted=0    spawn_value=4.54545   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow    obj;item;gun_upgrade;guns    blacklisted=1    spawn_value=1.92308   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
 /obj/item/weapon/gun_upgrade/mechanism/weintraub    obj;item;gun_upgrade;guns    blacklisted=0    spawn_value=3.125   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
+/obj/item/weapon/gun_upgrade/mechanism/funswitch    obj;item;gun_ungrade;guns    blacklisted=0    spawn_value=3.125   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
```
Comment (Jamini) [inline strings/loot_data/all_spawn_data.txt:6203]: This is an automatically generated file and doesn't need to be changed for this PR.

File: strings/loot_data/all_spawn_paths.txt [hunk 3]
```diff
@@ -1475,6 +1475,7 @@
 /obj/item/weapon/gun_upgrade/mechanism/overshooter
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow
 /obj/item/weapon/gun_upgrade/mechanism/weintraub
+/obj/item/weapon/gun_upgrade/mechanism/funswitch
```
Comment (Jamini) [inline strings/loot_data/all_spawn_paths.txt:1478]: This is an automatically generated file and doesn't need to be changed for this PR.

File: strings/loot_data/tags/gun_upgrade.txt [hunk 4]
```diff
@@ -13,6 +13,7 @@
 /obj/item/weapon/gun_upgrade/mechanism/overshooter    blacklisted=0    spawn_value=4.54545   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow    blacklisted=1    spawn_value=1.92308   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
 /obj/item/weapon/gun_upgrade/mechanism/weintraub    blacklisted=0    spawn_value=3.125   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
+/obj/item/weapon/gun_upgrade/mechanism/funswitch    blacklisted=0    spawn_value=3.125   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
```
Comment (Jamini) [inline strings/loot_data/tags/gun_upgrade.txt:16]: This is an automatically generated file and doesn't need to be changed for this PR.

File: strings/loot_data/tags/guns.txt [hunk 5]
```diff
@@ -125,6 +125,7 @@
 /obj/item/weapon/gun_upgrade/mechanism/overshooter    blacklisted=0    spawn_value=4.54545   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow    blacklisted=1    spawn_value=1.92308   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
 /obj/item/weapon/gun_upgrade/mechanism/weintraub    blacklisted=0    spawn_value=3.125   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
+/obj/item/weapon/gun_upgrade/mechanism/funswitch    blacklisted=0    spawn_value=3.125   spawn_price=100   prob_accompanying_obj=100    accompanying_objs=nothing
```
Comment (Jamini) [inline strings/loot_data/tags/guns.txt:128]: This is an automatically generated file and doesn't need to be changed for this PR.

File: strings/loot_data/tags/item.txt [hunk 6]
```diff
@@ -1990,7 +1990,8 @@
 /obj/item/weapon/gun_upgrade/trigger/boom    blacklisted=1    spawn_value=10    price_tag=100   100    
 /obj/item/weapon/gun_upgrade/mechanism/overshooter    blacklisted=0    spawn_value=5    price_tag=100   100    
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow    blacklisted=1    spawn_value=2    price_tag=100   100    
-/obj/item/weapon/gun_upgrade/mechanism/weintraub    blacklisted=0    spawn_value=3.33333    price_tag=100   100    
+/obj/item/weapon/gun_upgrade/mechanism/weintraub    blacklisted=0    spawn_value=3.33333    price_tag=100   100
```
Comment (Jamini) [inline strings/loot_data/tags/item.txt:1993]: This is an automatically generated file and doesn't need to be changed for this PR.

File: strings/loot_data/tags/obj.txt [hunk 7]
```diff
@@ -1991,6 +1991,7 @@
 /obj/item/weapon/gun_upgrade/mechanism/overshooter    blacklisted=0    spawn_value=5    price_tag=100   100    
 /obj/item/weapon/gun_upgrade/mechanism/glass_widow    blacklisted=1    spawn_value=2    price_tag=100   100    
 /obj/item/weapon/gun_upgrade/mechanism/weintraub    blacklisted=0    spawn_value=3.33333    price_tag=100   100    
+/obj/item/weapon/gun_upgrade/mechanism/funswitch    blacklisted=0    spawn_value=3.33333    price_tag=100   100
```
Comment (Jamini) [inline strings/loot_data/tags/obj.txt:1994]: This is an automatically generated file and doesn't need to be changed for this PR.

File: zzzz_modular_occulus/code/modules/projectiles/guns/mods/mods.dm [hunk 8]
```diff
@@ -16,4 +16,20 @@
...
+	desc = "The Solar ATF considers this \"fun switch\" to be the galaxy's smallest machine gun, so don't order one online unless you're lonely and really want to hang out with an SATF squad."
+	icon_state = "Funswitch"
+	rarity_value = 30
+
+/obj/item/gun_upgrade/mechanism/funswitch/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(
+		GUN_UPGRADE_FULLAUTO = TRUE,	//Makes the Paco able to fire full auto when installed
+		GUN_UPGRADE_RECOIL = 2		//But also makes it horrific to control
```
Comment (Jamini) [inline zzzz_modular_occulus/code/modules/projectiles/guns/mods/mods.dm:32]: Is this enough? Did you load up a server and test the cone on this? Did you test it with other mods like ergo grips?

File: strings/loot_data/all_spawn_data.txt (hunk 1 above)
Comment (Takykardi) [reply strings/loot_data/all_spawn_data.txt]: Should hopefully be removed from the PR.

File: strings/loot_data/all_spawn_paths.txt (hunk 3 above)
Comment (Takykardi) [reply strings/loot_data/all_spawn_paths.txt]: Should hopefully be removed from the PR.

File: strings/loot_data/tags/guns.txt (hunk 5 above)
Comment (Takykardi) [reply strings/loot_data/tags/guns.txt]: Should hopefully be removed from the PR.

File: strings/loot_data/tags/gun_upgrade.txt (hunk 4 above)
Comment (Takykardi) [reply strings/loot_data/tags/gun_upgrade.txt]: Should hopefully be removed from the PR.

File: strings/loot_data/tags/item.txt (hunk 6 above)
Comment (Takykardi) [reply strings/loot_data/tags/item.txt]: Should hopefully be removed from the PR.

File: strings/loot_data/tags/obj.txt (hunk 7 above)
Comment (Takykardi) [reply strings/loot_data/tags/obj.txt]: Should hopefully be removed from the PR.

File: zzzz_modular_occulus/code/modules/projectiles/guns/mods/mods.dm (hunk 8 above)
Comment (Takykardi) [reply zzzz_modular_occulus/code/modules/projectiles/guns/mods/mods.dm]: Values updated as requested in coding discussion. Newest test shows ~90 degree cone of fire by the end of a normal magazine burst.

**Comment by Shadow-Quill** [comment]
integration test failing for unrelated reason

**Review by Shadow-Quill — APPROVED** [review]

**Pull request merged**